node/src/types/block.rs

```diff
@@ -1741,6 +1741,12 @@ impl Block {
         })
     }
 
+    pub(crate) fn mark_after_emergency_upgrade(&mut self) {
+        // accumulated seed being equal to the parent hash will mean that the block is an immediate
+        // switch block created right after an emergency upgrade
+        self.header.accumulated_seed = *self.header.parent_hash.inner();
```

Comment:
This means we should update the `Block::hash` field I think.

Reply:
Right, I missed that! :scream:

Reply:
Good catch!